client/app/components/ui/modal.hbs

```diff
@@ -0,0 +1,49 @@
+{{#if @open}}
+  <EmberWormhole @to={{if @container @container "reveal-modal-container"}}>
+    <div class="reveal-overlay">
+      <div
+        class="reveal-overlay-target"
+        role="button"
+        {{on "click" @toggle}}
+        data-test-reveal-overlay
+      />
+      <div
+        class="reveal"
+        role="dialog"
+        aria-hidden="false"
+        tabindex="-1"
+        data-test-reveal-modal
+      >
+        {{yield to="message"}}
+
+        <hr>
+        <div class="grid-x">
+          <div class="cell large-6 small-padding-right">
+            <button
+              type="button"
+              class="button expanded large secondary"
+              {{on "click" @toggle}}
+            >
+              Continue Editing
+            </button>
+          </div>
+          <div class="cell large-6 small-padding-left">
+            {{yield @toggle to="action"}}
+          </div>
+        </div>
```

Comment:
This new component is named `Ui::Modal` but its hardcoded content makes it specifically an "[action]/continue editing" modal. If we need a modal for other purposes, we'd have to refactor this. 

We could just assume this component is specifically for the form edit confirmation modal, and when we need another modal we'll rename this one, e.g. `Ui::EditConfirmationModal` and make another `Ui::WhateverModal`. 

Or, should this modal be more generic? The confirm/cancel buttons could just be passed with the `:message`. But then, named blocks wouldn't be getting us much in a generic modal component, since we could just pass the `:message`, `:action`/buttons, and `:footer` in one block.

I'm fine either way. This works for now. I just wanted to point out that this isn't a generic modal component.

Reply:
@andycochran yep, I like your idea of splitting this out further — modal should be more generic, and we could have another component that uses that modal. I'll do that. Thanks.

Reply:
Going off of what @andycochran said about this being more generic. I think that our modals will most likely always have a button similar to (although might not be exactly the same) the `continue editing` button. The user is always able to toggle the modal through 1) a button similar to `Cancel` or `Continue Editing` or 2) the Close Modal button. So I think while it might not always have to say `continue editing`, I think we should keep this button here and the only thing that should be changeable would be the text on the button (e.g. `continue editing` or `return to xyz` etc.)

(also I really like that we have a passable action here for this button too)